### PR TITLE
Cs comments commondbtm/commonglpi

### PR DESCRIPTION
**Pull request by trasher**
Checked with coding-standards 0.5; but not activated in project, since there is a very huge job to do before.

### CODE REVIEW

File: inc/commondbtm.class.php [hunk 1]
```diff
@@ -226,6 +226,8 @@ function post_getFromDB() {
     * Actions done to not show some fields when geting a single item from API calls
     *
     * @param array $fields Fields to unset undiscloseds
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: Shouldn't it be &$fields to reflect the var reference ?

File: inc/commondbtm.class.php [hunk 2]
```diff
@@ -234,9 +236,9 @@ static public function unsetUndisclosedFields(&$fields) {
...
     * Retrieve all items from the database
     *
-    * @param $condition    condition used to search if needed (empty get all) (default '')
-    * @param $order        order field if needed (default '')
-    * @param $limit        limit retrieved datas if needed (default '')
+    * @param string $condition condition used to search if needed (empty get all) (default '')
+    * @param string $order     order field if needed (default '')
+    * @param string $limit     limit retrieved datas if needed (default '')
     *
     * @return all retrieved data in a associative array by id
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: No return type

File: inc/commondbtm.class.php [hunk 3]
```diff
@@ -340,8 +342,8 @@ function getLogTypeID() {
    /**
     * Update the item in the database
     *
-    * @param $updates      fields to update
-    * @param $oldvalues    array of old values of the updated fields
+    * @param array $updates   fields to update
+    * @param array $oldvalues array of old values of the updated fields
     *
     * @return nothing
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: void should be better ? (we haven't set it on the doc)

File: inc/commondbtm.class.php [hunk 4]
```diff
@@ -340,8 +342,8 @@ function getLogTypeID() {
    /**
     * Update the item in the database
     *
-    * @param $updates      fields to update
-    * @param $oldvalues    array of old values of the updated fields
+    * @param array $updates   fields to update
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: string[] ?

File: inc/commondbtm.class.php [hunk 5]
```diff
@@ -479,7 +481,7 @@ function restoreInDB() {
    /**
     * Mark deleted or purge an item in the database
     *
-    * @param $force force the purge of the item (not used if the table do not have a deleted field)
+    * @param boolean $force force the purge of the item (not used if the table do not have a deleted field)
     *               (default 0)
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: default FALSE ?

**Comment by trasher** [comment]
Thanks!

File: inc/commondbtm.class.php [hunk 6]
```diff
@@ -479,7 +481,7 @@ function restoreInDB() {
...
     *
     * @return true if succeed else false
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: @return boolean TRUE if succeed...

File: inc/commondbtm.class.php [hunk 7]
```diff
@@ -2395,7 +2424,7 @@ function showFormHeader($options=array()) {
     * is the parameter ID must be considered as new one ?
     * Default is empty of <0 may be overriden (for entity for example)
     *
-    * @param $ID ID of the item (-1 if new item)
+    * @param integert $ID ID of the item (-1 if new item)
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: -t

File: inc/commondbtm.class.php [hunk 8]
```diff
@@ -2582,7 +2611,7 @@ function checkEntity($recursive=false) {
    /**
     * Check global right on an object
     *
-    * @param $right Right to check : c / r / w / d
+    * @param mixed $right Right to check : c / r / w / d
     *
     * @return nothing
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: void ?

File: inc/commondbtm.class.php [hunk 9]
```diff
@@ -2600,7 +2629,7 @@ function checkGlobal($right) {
    /**
     * Get global right on an object
     *
-    * @param $right Right to check : c / r / w / d / READ / UPDATE / CREATE / DELETE
+    * @param mixed $right Right to check : c / r / w / d / READ / UPDATE / CREATE / DELETE
     *
     * @return nothing
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: void ?

File: inc/commondbtm.class.php [hunk 10]
```diff
@@ -3298,7 +3335,7 @@ static function getMassiveActionsForItemtype(array &$actions, $itemtype, $is_del
     *
     * @since version 0.85
     *
-    * @param $ma the current massive action object
+    * @param MassiveAction $ma the current massive action object
     *
     * @return false if parameters displayed ?
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: boolean

File: inc/commondbtm.class.php [hunk 11]
```diff
@@ -3873,9 +3914,9 @@ static function getItemEntity($itemtype, $items_id) {
...
     * @since version 0.83
     *
-    * @param $field     String         name of the field
-    * @param $values    String/Array   with the value to display or a Single value
-    * @param $options   Array          of options
+    * @param string       $field   name of the field
+    * @param string|array $values  with the value to display or a Single value
+    * @param array        $options Array of options
     *
     * @return return the string to display
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: string

File: inc/commondbtm.class.php [hunk 12]
```diff
@@ -3685,8 +3724,8 @@ function getUnicityErrorMessage($msgs, $unicity, $doubles) {
    /**
     * Check field unicity before insert or update
     *
-    * @param add                 true for insert, false for update (false by default)
-    * @param $options   array
+    * @param boolean $add     true for insert, false for update (false by default)
+    * @param array   $options array
     *
     * @return true if item can be written in DB, false if not
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: boolean

File: inc/commondbtm.class.php [hunk 13]
```diff
@@ -3579,8 +3616,8 @@ function filterValues($display=true) {
    /**
     * Add more check for values
     *
-    * @param $datatype   datatype of the value
-    * @param &$value     value to check (pass by reference)
+    * @param string $datatype datatype of the value
+    * @param array  $value    value to check (pass by reference)
     *
     * @return true if value is ok, false if not
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: boolean

File: inc/commondbtm.class.php [hunk 14]
```diff
@@ -3433,9 +3470,9 @@ function getOptions() {
...
     * @since version 0.83
     *
-    * @param $field   the field in which looking for the value (for example : table, name, etc)
-    * @param $value   the value to look for in the field
-    * @param $table   the table (default '')
+    * @param string $field the field in which looking for the value (for example : table, name, etc)
+    * @param string $value the value to look for in the field
+    * @param string $table the table (default '')
     *
     * @return then search option id, or -1 if not found
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: mixed

File: inc/commondbtm.class.php [hunk 15]
```diff
@@ -3388,9 +3425,9 @@ static function dropdown($options=array()) {
...
     * Return a search option by looking for a value of a specific field and maybe a specific table
     *
-    * @param $field   the field in which looking for the value (for example : table, name, etc)
-    * @param $value   the value to look for in the field
-    * @param $table   the table (default '')
+    * @param string $field the field in which looking for the value (for example : table, name, etc)
+    * @param string $value the value to look for in the field
+    * @param string $table the table (default '')
     *
     * @return then search option array, or an empty array if not found
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: array ?

File: inc/commondbtm.class.php [hunk 16]
```diff
@@ -3344,7 +3381,7 @@ function getForbiddenStandardMassiveAction() {
     *
     * This should be overloaded in Class
     *
-    * @param $checkitem link item to check right   (default NULL)
+    * @param object $checkitem link item to check right (default NULL)
     *
     * @return an array of massive actions
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: array

File: inc/commondbtm.class.php [hunk 17]
```diff
@@ -3312,9 +3349,9 @@ static function showMassiveActionsSubForm(MassiveAction $ma) {
...
     * @since version 0.85
     *
-    * @param $ma the current massive action object
-    * @param $item the item on which apply the massive action
-    * @param $ids an array of the ids of the item on which apply the action
+    * @param MassiveAction $ma   the current massive action object
+    * @param CommonDBTM    $item the item on which apply the massive action
+    * @param array         $ids  an array of the ids of the item on which apply the action
     *
     * @return nothing (direct submit to $ma object)
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: void ?

File: inc/commondbtm.class.php [hunk 18]
```diff
@@ -4096,10 +4137,10 @@ function getValueToDisplay($field_id_or_search_options, $values, $options=array(
...
-    * @param $field     String         name of the field
-    * @param $name      string         name of the select (if empty use linkfield) (default '')
-    * @param $values    String/Array   with the value to select or a Single value (default '')
-    * @param $options   Array          of options
+    * @param string       $field   name of the field
+    * @param string       $name    name of the select (if empty use linkfield) (default '')
+    * @param string|array $values  with the value to select or a Single value (default '')
+    * @param array        $options aArray of options
     *
     * @return return the string to display
```
Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: string

File: inc/commondbtm.class.php (hunk 1 above)
Comment (trasher) [inline inc/commondbtm.class.php]: This does not match CS rules ;)

Comment (AnaelMobilia) [inline inc/commondbtm.class.php]: Yes, of course !
But couldn't documented it (and maybe update CS too) ?

I think it's an important information for a dev.

Your opinion ?

File: inc/commondbtm.class.php (hunk 5 above)
Comment (trasher) [inline inc/commondbtm.class.php]: No, default is set to 0

File: inc/commondbtm.class.php (hunk 7 above)
Comment (trasher) [inline inc/commondbtm.class.php]: :D

File: inc/commondbtm.class.php (hunk 1 above)
Comment (trasher) [inline inc/commondbtm.class.php]: The method signature is visible in the doc, so no need to add the `&` in the docblock (current api doc is not up-to-date, I've requested the update, I'll give you the link to the result once done).

**Pull request merged**